idaes/apps/caprese/plotlibrary.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+
+import pandas as pd
+import matplotlib.pyplot as plt
+from pyomo.environ import (
+    ComponentUID,
+    Reference,
+    )
+
+__author__ = "Kuan-Han Lin"
+
+
+class PLANT_PlotLibrary(object):
```

Comment:
Yeah, inheritance is the main reason. When I wrote these codes, I wanted the plotting things to be one of the functions of  the data manger, so the user can plot by calling the data manager. However, I have to say I'm not sure if it's a good idea. Let's talk about this!

Reply:
If we aren't caching data on the objects, I say we just have utility functions for plotting. It will be simpler to read and call. I think the functions in this file are good otherwise.